### PR TITLE
230 feedback endpoint

### PR DESCRIPTION
**Pull request by abdul-shajin**
Added feedback model with fields Name and Message
New table feedbacks
Almost same as Ticket table without email
Added simple test cases for model validity
Added fixtures
Added Mutation file for feedback
Added FeedbackTypes
Added mutation type addFeedback
Test cases for new mutation actions
Added Elm code to call the new endpoint if email is not present in feedback form

### CODE REVIEW

File: db/migrate/20180723094927_create_feedbacks.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateFeedbacks < ActiveRecord::Migration[5.2]
+  def change
+    create_table :feedbacks, id: :uuid do |t|
+      t.string :name
+      t.text :message
+      t.uuid :article_id, null: true
```
Comment (neerajdotname) [inline db/migrate/20180723094927_create_feedbacks.rb]: @abdul-shajin why null true?

File: db/migrate/20180723094927_create_feedbacks.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateFeedbacks < ActiveRecord::Migration[5.2]
+  def change
+    create_table :feedbacks, id: :uuid do |t|
+      t.string :name
+      t.text :message
```
Comment (neerajdotname) [inline db/migrate/20180723094927_create_feedbacks.rb]: Please make message not nullable. There is no point getting an empty feedback.

File: db/migrate/20180723094927_create_feedbacks.rb (hunk 1 above)
Comment (neerajdotname) [reply db/migrate/20180723094927_create_feedbacks.rb]: Please add both foreign_key and and index on `article_id`. Ref https://github.com/bigbinary/acehelp/blob/master/db/migrate/20180724114503_add_missing_foreign_keys.rb for more.

File: app/javascript/Client/Data/ContactUs.elm [hunk 3]
```diff
@@ -82,8 +82,26 @@ addTicketMutation =
...
+            GQLBuilder.extract <|
+                GQLBuilder.field "addFeedback"
+                    [ ( "input"
+                      , Arg.object
+                          [ ("name", Arg.variable guestNameVar )
+                          , ("message", Arg.variable guestMessageVar )
+                          ]
+                      )
+                    ]
+                    errorsExtractor
```
Comment (breezyboa) [inline app/javascript/Client/Data/ContactUs.elm]: A feedback is associated with an article. The mutation should also take in article id as refrence.
Also move this to `Data/Article.elm`

File: app/javascript/Client/Request/ContactUs.elm [hunk 4]
```diff
@@ -29,3 +29,11 @@ requestAddTicketMutation feedbackFrom =
...
                 GQLBuilder.request feedbackFrom addTicketMutation
         )
+
+requestAddFeedbackMutation : FeedbackForm -> Reader ( NodeEnv, ApiKey ) (Task GQLClient.Error (Maybe (List GQLError)))
+requestAddFeedbackMutation feedbackFrom =
+    Reader.Reader
+        (\( env, apiKey ) ->
+            GQLClient.sendMutation (graphqlUrl env) <|
+                GQLBuilder.request feedbackFrom addFeedbackMutation
+        )
```
Comment (breezyboa) [inline app/javascript/Client/Request/ContactUs.elm]: Move this to `Request/Article.elm`
Add a new at eof

File: db/migrate/20180723094927_create_feedbacks.rb (hunk 1 above)
Comment (abdul-shajin) [reply db/migrate/20180723094927_create_feedbacks.rb]: some feedbacks can be not based on articles, 
eg: contact us tab.. so can be null

Comment (abdul-shajin) [reply db/migrate/20180723094927_create_feedbacks.rb]: my bad, from contact us, there's a validation for email which will never go to feedback tables. will make the change.

**Comment by neerajdotname** [comment]
Feedback must be associated with an article .

Ticket must have an email.

**Comment by neerajdotname** [comment]
@abdul-shajin _a Assigning it to the right person.

**Comment by abdul-shajin** [comment]
I had updated migration file according to the review suggestions since no-one has migrated to their system. 
I hope thats okay. @breezyboa

**Comment by breezyboa** [comment]
@abdul-shajin _a can you resolve conflicts

**Comment by breezyboa** [comment]
@abdul-shajin _a can you check why builds are failing.

**Comment by neerajdotname** [comment]
@abdul-shajin Reminder that this PR is still in your bucket. Please do the necessary to get this PR merged.

**Comment by abdul-shajin** [comment]
CircleCI is asking to run db:migrate on TEST environment. 
I am quite not sure how to do that. Can you have a look @vipulnsward

**Comment by prathamesh-sonpatki** [comment]
@abdul-shajin 

``` 
Migrations are pending. The schema.rb is not updated. To resolve this issue, run:

        bin/rails db:migrate RAILS_ENV=test
```

Please update the schema.rb manually and check that in. Then circle ci won't complain.

**Comment by vipulnsward** [comment]
@abdul-shajin _a

**Comment by abdul-shajin** [comment]
@breezyboa _a

**Pull request merged**